autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ScalingRequest.java

```diff
@@ -81,35 +85,47 @@
     @Inject
     private UsageReportingService usageReportingService;
 
-    public ScalingRequest(Cluster cluster, ScalingPolicy policy, int totalNodes, int desiredNodeCount, List<String> decommissionNodeIds) {
+    @Inject
+    private LimitsConfigurationService limitsConfigurationService;
+
+    public ScalingRequest(Cluster cluster, ScalingPolicy policy, int existingClusterNodeCount, int existingHostGroupNodeCount,
+            int desiredHostGroupNodeCount, List<String> decommissionNodeIds) {
         this.cluster = cluster;
         this.policy = policy;
-        this.totalNodes = totalNodes;
-        this.desiredNodeCount = desiredNodeCount;
+        this.existingClusterNodeCount = existingClusterNodeCount;
+        this.existingHostGroupNodeCount = existingHostGroupNodeCount;
+        this.desiredHostGroupNodeCount = desiredHostGroupNodeCount;
         this.decommissionNodeIds = decommissionNodeIds;
     }
 
     @Override
     public void run() {
         LoggingUtils.buildMdcContext(cluster);
         try {
-            int scalingAdjustment = desiredNodeCount - totalNodes;
+            int scalingAdjustment = desiredHostGroupNodeCount - existingHostGroupNodeCount;
             if (!decommissionNodeIds.isEmpty()) {
                 scaleDownByNodeIds(decommissionNodeIds);
             } else if (scalingAdjustment > 0) {
-                scaleUp(scalingAdjustment, totalNodes);
+                int cbSupportedMaxClusterSize = limitsConfigurationService.getMaxNodeCountLimit();
```

Comment:
Asked earlier also. Looks like the check is in place... 
Goes to the second question: Should the check be in place? (If CB reduces the limit - should the behaviour be to fail (user expectations not met) vs scale to the reduced limit.
This gets even more complicated since the same scenario can be hit if a different hostGroup is resized up (without the CB limit changing).

Reply:
answered above also. just before scale-up is requested, node count is adjusted for max cluster size.